src/Lib/Element.php

```diff
@@ -108,18 +109,43 @@ public function hasOptions()
     }
 
     /**
+     * Returns whether this element conatins elements
+     * @return bool
+     */
+    public function hasElements()
+    {
+        return count($this->elements) > 0;
+    }
+
+    /**
+     * Returns whether this object has anything to serialise
+     * @return bool
+     */
+    public function isEmpty()
+    {
+        return !$this->hasOptions() && !$this->hasElements();
+    }
+
+
+    /**
      * Extract any external elements from the elements and combine them with the elements array
      * @return array
      */
     protected function extractElements()
     {
         $elements = [];
+
+        // Fetch external elements from the elements
         foreach($this->elements as $element) {
             $elements = array_merge($elements, $element->getExternalElements());
         }
+
+        //Merge them with our elements
         $elements = array_merge($this->elements, $elements);
```

Comment:
Not sure what this is doing... you're merging $this->elements with an unravelled version of itself?